### ISSUE TITLE
addLoggedUser while in global context

### ISSUE THREAD

**Issue by ChiKaLiO** [WelingtonMonteiro/mongoose-history-trace#2]
How to define LoggedUser while using the plugin globally?

**Comment by WelingtonMonteiro** [comment]
Use normally as an example, when adding it globally the "addLoggedUser" method will be available for all collections.

https://www.npmjs.com/package/mongoose-history-trace#--addloggeduser-object--required

**Comment by ChiKaLiO** [comment]
I'm using rest api so no session in the server, is there a way to extract user from req every time ?

**Comment by WelingtonMonteiro** [comment]
I don't know if I understood correctly, but in the example of the code I always extract the user from the request, in the case of the example using npm express, but depending on the npm used, instead of req.user, context.user is used.
And in the example I don't use the session, the user is coming from the request, assuming it was placed by some midleware manually.
Example of a controller:

```
async function update (req, res) {
    const user = req.user   //< -- GET user on request
    ...
 
    Model.addLoggedUser(user)   //<-- SET LOGGED USER in context
    
    const result = await Model.findByIdAndUpdate(query, {$set: mod})
 
    return res.send(result)    
}
```

**Comment by ChiKaLiO** [comment]
> I don't know if I understood correctly, but in the example of the code I always extract the user from the request, in the case of the example using npm express, but depending on the npm used, instead of req.user, context.user is used.
> And in the example I don't use the session, the user is coming from the request, assuming it was placed by some midleware manually.
> Example of a controller:
> 
> ```
> async function update (req, res) {
>     const user = req.user   //< -- GET user on request
>     ...
>  
>     Model.addLoggedUser(user)   //<-- SET LOGGED USER in context
>     
>     const result = await Model.findByIdAndUpdate(query, {$set: mod})
>  
>     return res.send(result)    
> }
> ```

yes exactly, but this way i need to do it for every route :(

**Comment by WelingtonMonteiro** [comment]
Yes, I currently haven't implemented any global middleware to pick up users, because I haven't figured out a way to do it globally and automatically. for this I need to ensure that there will be a user in Sessions, and also ensure that it is the correct user and not from another request. I left this implementation for the future. Because I need to save and request in some Sessions and can read them in the mongoose plugin middleware, and I didn't find anything that works.
